rust/src/lib/route.rs

```diff
@@ -78,6 +78,22 @@ impl Routes {
                     error!("{}", e);
                     return Err(e);
                 }
+                if !route.destination.as_deref().unwrap_or_default().is_empty() {
+                    let destination_address = route.destination.as_deref().unwrap_or_default();
+                    if destination_address.ip.to_str() == "0.0.0.0"
+                        && destination_address.prefix_length >= 8 {
+                        let e = NmstateError::new(
+                            ErrorKind::InvalidArgument,
+                            format!(
+                                "0.0.0.0/8 and its subnet cannot be used as \
+                            the route destination: {:?}",
```

Comment:
Normally we do not show debug output to user as error message.

Meanwhile, provide a hint to user that they might want to use default gateway `0.0.0.0/0`.